### PR TITLE
Added wallet-selector with NEAR Wallet and Sender supported

### PR DESCRIPTION
**Pull request by mattlockyer**
**Known issue:**
- Sender wallet logs out when you reload the page
- To mitigate, after a sender tx, the page should update immediately without needing to reload

**Changelog:**

- pretty major refactor to `./utils.ts` and `./store/wallet.ts`
- added `./utils/wallet-selector-compat.ts` which has most of the utility methods needed to use the wallet-selector

- `./utils.ts` getBurrow method now returns some extra utilities such as `fetchData` and `hideModal` used by `./store/wallet.ts` method `executeMultipleTransactions` in the event the wallet is "Sender" and there was no redirect, the modal must be closed and new data for the account must be fetched

- `./store/wallet.ts` the entire BatchWallet class and wallet connection has been removed in favor of wallet-selector `signAndSendTransactions` which requires custom transaction actions found here: https://github.com/near/wallet-selector/blob/dev/packages/core/src/lib/wallet/actions.ts 

- uses wallet-selector alpha 3 which is a VERY fresh release
- still potentially some updates coming to address the Sender wallet issue mentioned above (e.g. stay signed in even when page reloads until user signs out)

**Upcoming:**
- adding support to switch accounts in Sender from the UI, when Sender wallet supports this... as it's already in wallet-selector

**Tested:**

NEAR Wallet
- sign in
- deposit
- borrow
- repay
- sign out

Sender Wallet
- sign in
- deposit
- borrow
- repay
- sign out

### CODE REVIEW

File: src/assets/near-icon.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 288 288"><g id="Layer_1" data-name="Layer 1"><path d="M187.58,79.81l-30.1,44.69a3.2,3.2,0,0,0,4.75,4.2L191.86,103a1.2,1.2,0,0,1,2,.91v80.46a1.2,1.2,0,0,1-2.12.77L102.18,77.93A15.35,15.35,0,0,0,90.47,72.5H87.34A15.34,15.34,0,0,0,72,87.84V201.16A15.34,15.34,0,0,0,87.34,216.5h0a15.35,15.35,0,0,0,13.08-7.31l30.1-44.69a3.2,3.2,0,0,0-4.75-4.2L96.14,186a1.2,1.2,0,0,1-2-.91V104.61a1.2,1.2,0,0,1,2.12-.77l89.55,107.23a15.35,15.35,0,0,0,11.71,5.43h3.13A15.34,15.34,0,0,0,216,201.16V87.84A15.34,15.34,0,0,0,200.66,72.5h0A15.35,15.35,0,0,0,187.58,79.81Z"/></g></svg>
```
Comment (ciocan) [inline src/assets/near-icon.svg:1]: we already have the near icon in 
components/TokenIcon

File: src/components/Header/Menu.tsx [hunk 2]
```diff
@@ -59,24 +67,27 @@ export const HamburgerMenu = ({ anchorEl, setAnchorEl }: Props) => {
         "aria-labelledby": "logout-button",
       }}
     >
-      {accountId && (
-        <MenuItem sx={{ backgroundColor: "white" }} onClick={handleClaimAll}>
+      {accountId && [
```
Comment (ciocan) [inline src/components/Header/Menu.tsx:70]: nice one, if forgot can use []

File: src/components/Header/WalletButton.tsx [hunk 3]
```diff
@@ -31,13 +65,27 @@ const WalletButton = () => {
...
+          sx={{
+            marginTop: 0.6,
+            fontSize: "0.8rem",
+            display: "flex",
+            flexFlow: isMobile ? "column" : "row",
+          }}
+        >
+          <div style={{ fontWeight: "bold" }}>{accountTrim(accountId)}</div>
+          <div style={{ display: "flex", justifyContent: "flex-end", marginLeft: 4 }}>
+            {balance.substring(0, balance.length - 2)}
```
Comment (ciocan) [inline src/components/Header/WalletButton.tsx:84]: this does not take into account balances without decimals, like 1 NEAR
you can make a new TOKEN_FORMAT with only 2 decimals

File: src/store/wallet.ts [hunk 4]
```diff
@@ -1,120 +1,69 @@
...
-        ),
-      });
-    }),
-  );
 
-  return walletConnection.requestSignTransactions({
-    transactions: nearTransactions,
-    callbackUrl,
-  })!;
+  // console.log(res)
```
Comment (ciocan) [inline src/store/wallet.ts:63]: remove commented

File: src/utils.ts [hunk 5]
```diff
@@ -9,49 +9,78 @@ import {
...
 
 export const isTestnet = getConfig(defaultNetwork).networkId === "testnet";
 
-let burrow: IBurrow;
-
-const nearTokenIds = {
-  mainnet: "wrap.near",
-  testnet: "wrap.testnet",
-};
+// interface AccountChangeFunction {
```
Comment (ciocan) [inline src/utils.ts:20]: remove comments

File: src/utils.ts [hunk 6]
```diff
@@ -9,49 +9,78 @@ import {
...
+export const getBurrow = async ({
+  fetchData,
+  hideModal,
+  signOut,
+}: GetBurrowArgs = {}): Promise<IBurrow> => {
+  if (burrow && !resetBurrow) return burrow;
+  resetBurrow = false;
+
+  const changeAccount = async (accountId) => {
+    console.log("account changed", accountId);
```
Comment (ciocan) [inline src/utils.ts:59]: remove console

File: src/utils.ts [hunk 7]
```diff
@@ -138,24 +165,24 @@ export async function initContract(): Promise<IBurrow> {
...
-export async function login(walletConnection: WalletConnection) {
-  // Allow the current app to make calls to the specified contract on the
-  // user's behalf.
-  // This works by creating a new access key for the user's account and storing
-  // the private key in localStorage.
-  await walletConnection.requestSignIn({
-    contractId: LOGIC_CONTRACT_NAME,
-  });
-}
+// export function logout(walletConnection: WalletConnection) {
```
Comment (ciocan) [inline src/utils.ts:168]: remove comments

File: src/utils/wallet-selector-compat.ts [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  }
+  return near;
+};
+
+export const getAccount = async (viewAsAccountId: string | null) => {
+  near = getNear();
+  return new Account(near.connection, viewAsAccountId || accountId);
+};
+
+// export const viewFunction = async ({ contractId, methodName, args }: WalletMethodArgs) => {
```
Comment (ciocan) [inline src/utils/wallet-selector-compat.ts:78]: remove comments

**Review by ciocan — APPROVED** [review]
LGTM, only a few comments with minor changes

File: src/assets/near-icon.svg (hunk 1 above)
Comment (mattlockyer) [reply src/assets/near-icon.svg:1]: this one has circle with white, I think N black svg transparent is better for currency symbol

File: src/components/Header/WalletButton.tsx (hunk 3 above)
Comment (mattlockyer) [reply src/components/Header/WalletButton.tsx:84]: great point!

Comment (mattlockyer) [reply src/components/Header/WalletButton.tsx:84]: gonna use: `Number.parseFloat(balance).toFixed(2)`

**Pull request merged**